### PR TITLE
ZK-3629: Indeterminate progressmeter

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 1]
```diff
@@ -48,10 +61,30 @@ zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
 			}
 		}
 	},
+	_handleIndeterminateAnimation: function () { // ZK-3629: for ie9 indetermination animation
+		if (zk.ie9) {
+			var $img = jq(this.$n('img'));
+			function loop() {
+				$img.css({left: '-100%', width: '50%'});
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]: ie9 css handle width

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 2]
```diff
@@ -48,10 +61,30 @@ zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
 			}
 		}
 	},
+	_handleIndeterminateAnimation: function () { // ZK-3629: for ie9 indetermination animation
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]: _handleIndeterminateAnimation: zk.ie9 ? function (init) {
//...
} : zk.$void,

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 3]
```diff
@@ -48,10 +61,30 @@ zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
...
 	},
+	_handleIndeterminateAnimation: function () { // ZK-3629: for ie9 indetermination animation
+		if (zk.ie9) {
+			var $img = jq(this.$n('img'));
+			function loop() {
+				$img.css({left: '-100%', width: '50%'});
+				$img.animate ({
+					left: '100%',
+				}, 1500, 'linear', function() {
+					loop();
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]: use interval

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes and merged, enjoy!

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 4]
```diff
@@ -19,6 +19,8 @@ it will be useful, but WITHOUT ANY WARRANTY.
  */
 zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
 	_value: 0,
+	_indeterminate: false,
+	_indeterminateAnimation : false,
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]:
```suggestion
	_indeterminateAnimation: false,
```

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 5]
```diff
@@ -40,18 +54,40 @@ zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
...
+		} else if (this._indeterminateAnimation) {
+			$img.stop(true);
+			this._indeterminateAnimation = false;
+			this._fixImgWidth();
+		}
+	} : zk.$void,
+	_startIndeterminateAnimation: function (target) { // ZK-3629: for ie9 indetermination animation
+		var self = this;
+		target.css({left: '-100%'});
+		target.animate ({
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]:
```suggestion
		target.animate({
```

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 6]
```diff
@@ -40,18 +54,40 @@ zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
...
+			left: '100%',
+		}, 1500, 'linear', function() {
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]:
```suggestion
		}, 1500, 'linear', function () {
```

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**